core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java

```diff
@@ -1042,6 +1046,81 @@ else if (thatNextRangeIndex == thatRangeCount) {
         return new SortedRangeSet(type, inclusive, blockBuilder.build(), unionIsDiscreteSet(that, resultRangeIndex > 0));
     }
 
+    SortedRangeSet linearDiscreteSetUnion(SortedRangeSet that)
+    {
+        return linearDiscreteSetUnion(this, that);
+    }
+
+    private SortedRangeSet linearDiscreteSetUnion(SortedRangeSet leftRangeSet, SortedRangeSet rightRangeSet)
+    {
+        int leftRangeCount = leftRangeSet.getRangeCount();
+        int rightRangeCount = rightRangeSet.getRangeCount();
+
+        boolean[] inclusive = new boolean[2 * (leftRangeCount + rightRangeCount)];
+        BlockBuilder blockBuilder = type.createBlockBuilder(null, 2 * (leftRangeCount + rightRangeCount));
+
+        int resultRangeIndex = 0;
+        int leftNextRangeIndex = 0;
+        int rightNextRangeIndex = 0;
+
+        int currentUnionStart = 0;
+        while (leftNextRangeIndex < leftRangeCount && rightNextRangeIndex < rightRangeCount) {
+            int compare = compareValues(
```

Comment:
you could add a comment that invariant is that value represented by `leftNextRangeIndex` is smaller than the next value on the right side, therefore we only add values from the left side, which allows us to add values in bulk

Reply:
Added a comment and please advise